File: daily/C++版本/682. 棒球比赛.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        int sum=0;
        stack<int> stack;
        for(int i=0;i<ops.size();i++){
            if(ops[i]=="C"){
                sum-=stack.top();
                stack.pop();
            }
            else if(ops[i]=="D"){
                sum+=stack.top()*2;
                stack.push(stack.top()*2);
            }
            else if(ops[i]=="+"){
                int a,b;    //保存弹栈一次和两次的值
                a=stack.top(),stack.pop();
                b=stack.top(),stack.pop();
                sum+=a+b;
                stack.push(b),stack.push(a),stack.push(a+b);
            }
            else{
                int a;
                stringstream ss;    //字符串转数字
                ss<<ops[i];
                ss>>a;
                sum+=a;
                stack.push(a);
            }
        }
        return sum;
    }
};

File: daily/栈/栈easy/682. 棒球比赛.js
/**
 * @param {string[]} ops
 * @return {number}
 */
var calPoints = function(ops) {
    let stack=[]
    ops.forEach(item=>{
        if(item=='+')
            stack.push(stack[stack.length-1] + stack[stack.length-2])
        else if(item=='D')
            stack.push(stack[stack.length-1]*2)
        else if(item=='C')
            stack.pop()
        else
            stack.push(Number(item))
    })
    return stack.reduce((sum,item)=>sum+=item)
};